cmd/pebble/test.go

```diff
@@ -329,6 +329,17 @@ func runTest(dir string, t test) {
 		close(workersDone)
 	}()
 
+	if maxSize > 0 {
+		go func() {
+			for {
+				time.Sleep(10 * time.Second)
+				if db.Metrics().DiskSpaceUsage() > maxSize*1e6 {
```

Comment:
I wonder if I should restrict this to Live Bytes by removing `ObseleteBytes`

Reply:
They are removed after each compaction as part of the "cleaning turn", so I'd expect files to stick around as obsolete for a very short and transient period of time.

Also have you looked into `db.opts.FS.GetDiskUsage` ? That might be better here, and it has the benefit of being more parallelizable (db.Metrics() grabs the db mutex). And once you make that change, you can also reduce the sleep interval down to 2 seconds or so instead of 10 seconds.

Reply:
ah hmm. Using `db.opts.FS.GetDiskDiskUsage` would be ideal; however, `opts` is a private field, so to implement this, I'd need to add another public method to `pebble.DB`, which actually already has a `EstimateDiskUsage()` public method that seems more complicated and expensive than what we need. How would you feel if added the following to the `pebble.DB` API:

```
func (d *DB) GetTotalDiskUsage() (uint64, error){
  return db.opts.FS.GetDiskUsage(d.dirname)
}
```

Reply:
I'd rather update `newPebbleDB` (which is only called here) to return both the db and the `*Options`, that way we can just grab the `FS` off the `Options`. Sounds simpler than updating the DB API for a test-only change.